sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,230 @@
+# Authors: Andrea Bravi <a.bravi@uottawa.ca>
+#          Nikolay Mayorov <n59_ru@hotmail.com>
+# License: 3-clause BSD
+
+from __future__ import division
+
+import numpy as np
+from scipy.sparse import issparse
+from scipy.special import digamma
+
+from ..base import BaseEstimator
+from .base import SelectorMixin
+from ..metrics.cluster.supervised import mutual_info_score
+from ..utils.validation import check_X_y
+from ..neighbors import NearestNeighbors
+
+
+def _compute_mi_cc(x, y, n_neighbors):
+    n_samples = x.size
+
+    x_std = np.std(x)
+    if x_std > 0:
+        x = x / x_std
```

Comment:
Seems like we shouldn't to it for sparse `X`: storing each row scaled can result in storing the whole `X` in dense format, which probably was avoided for a good reason.

As for dense `X` I think it's fine to introduce `copy_X=True` parameter and the either modify `X` in place or make a copy then modify.